### DISCUSSION TITLE
Welcome to MIDI-RtMidi-FFI Discussions!

### DISCUSSION THREAD

**Discussion by jbarrett** [jbarrett/MIDI-RtMidi-FFI#25] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

(Yes, this is the default welcome message ...)

**Comment by tux99** [comment]
Hi,
I have been writing MIDI/synthesizer related software in Perl for more than 10 years (in my spare time out of personal interest and therefore on and off, not regularly) primarily using MIDI::ALSA from Peter J Billam (R.I.P.) whom I'm forever grateful for creating this module as it enabled me to write MIDI software in Perl and then later also using  Win32API::MIDI to add support for Windows.

I only discovered MIDI::RtMidi::FFI recently and I'm quite excited about it as it should give the ability to add macOS support to my software.

I'm a Linux user since the early days (mid '90s) but I like adding support for other OSes in my software too as long as it's not too much of a headache.

**Comment by jbarrett** [comment]
Hey @tux99, welcome!

Yeah, this should offer Mac, Windows and Android(!) support, though I would point out that there are issues with the callback interface on Windows - funky stuff around threading which I haven't been able to pin down yet.

I believe the software has been used on MacOS with some success by @ology -- manly using MIDI-thru via IAC? Virtual devices *should* be supported though.

**Comment by ology** [comment]
Correct - On my mac with the IAC Device, which I thought **was** a virtual device?
